enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status string `json:"status,omitempty"`
+	// Errors is a list of gathering errors from the last gathering.
+	Errors []string `json:"errors,omitempty"`
+	// List of active gatherers in the last gathering
+	Gatherers []string `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low,omitempty"`
+	// Number of active Insights healthchecks with moderate severity
+	ModerateCount int `json:"moderate,omitempty"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important,omitempty"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical,omitempty"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total,omitempty"`
+}
+
+type SimpleContentAccessStatus struct {
+	// LastDownloadTime is the last time when the SCA certs were downloaded.
+	LastDownloadTime time.Time `json:"lastDownloadTime,omitempty"`
+	// Status provides last known status information of the SCA controller.
+	Status string `json:"status,omitempty"`
+}
+
+type ClusterTransferStatus struct {
+	// LastCheckTime is the last time when the ClusterTranfer was checked.
+	LastCheckTime time.Time `json:"lastCheckTime,omitempty"`
+	// Status provides last known status information of the ClusterTransfer controller.
+	Status string `json:"status,omitempty"`
+}
+
+```
+
+### User Stories
+
+- As an OCP user/cluster-admin I would like to be able to trigger on demand gathering (there were some active Insights health checks in the cluster, which should be fixed and I want to refresh the Insights status). 
+- As an OCP user/cluster-admin I would like to have a convenient way to configure Insights.
+- As an OCP user/cluster-admin I would like to see the latest Insights status.
+- As an OCP user/cluster-admin I would like to exclude/include particular gatherers.
+
+### API Extensions
+
+This is adding a new `insights.config.openshift.io` API. The proposal of the structure of the new API is mentioned in the [Proposal](#Proposal) section.
+
+### Risks and Mitigations
+
+Risk: Insights operator is not able to read the new config API.
+
+Mitigation: The Insights operator is marked as unvailable through the corresponding clusteroperator condition.
+
+Risk: Somebody/something is repeatedly updating the `ForceGatheringReason` attribute and thus forcing gathering. There is a risk that 
+the cluster would upload data too often, but this assumes that the "repeating interval" allows uploading (if it's forced too quickly then there will be no upload, because the gathering is not over).
+
+Mitigation: Maybe we should introduce some limit on the number of forced gatherings within 1 hour, for example. 
```

Comment:
how often are reports gathered outside of forced gatherings?

is there any desire for that interval to be configurable?

Reply:
and also "whenever a new Insights pod comes up".  Or at least, I think I've seen multiple Insights uploads from clusters that were rolling their control planes.  But this new structure will make it easy to persist the previous time between Insights pods, so maybe that start-time gather will go away.

Reply:
correct.